Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -488,10 +488,17 @@ void webkitWebViewBaseToplevelWindowIsActiveChanged(WebKitWebViewBase* webViewBa
 static void webkitWebViewBaseUpdateVisibility(WebKitWebViewBase* webViewBase)
 {
     WebKitWebViewBasePrivate* priv = webViewBase->priv;
-    const bool isVisible = gtk_widget_get_mapped(GTK_WIDGET(webViewBase))
-        && priv->toplevelOnScreenWindow->isInMonitor()
-        && !priv->toplevelOnScreenWindow->isMinimized()
-        && !priv->toplevelOnScreenWindow->isSuspended();
+    const bool isVisible = [&] {
+        if (!gtk_widget_get_mapped(GTK_WIDGET(webViewBase)))
+            return false;
+
+        if (!priv->toplevelOnScreenWindow)
+            return true;
```

Comment:
Why is this correct?

I would be tempted to think the opposite: if there is no top level on-screen window, then the web view is NOT visible.

Reply:
OK, then what I would do is: land this as-is, then follow up in a separate commit to change it to `return false`, don't backport the second commit, wait and see whether users complain that something breaks in 2.49.3.